### PR TITLE
Add Markdown export of crash logs

### PR DESCRIPTION
**Pull request by TobiGr**
Add content country as additional debug information to reports

#### What is it?
- [ ] Bug fix (user facing)
- [x] Feature (user facing)
- [ ] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Adds button to error screen to copy markdown crash report and button to open our GitHub issue page.

#### Fixes the following issue(s)
Part of #1816

#### Testing apk

[app-debug.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/4620341/app-debug.apk.zip)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by B0pol** [comment]
Can you send a screenshot / provide more info? Is it just [Crash report to markdown](Error Report to Markdown reporter
https://teamnewpipe.github.io/CrashReportToMarkdown/) but in-app?

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 1]
```diff
@@ -351,7 +385,8 @@ private String buildJson() {
                     .object()
                     .value("user_action", getUserActionString(errorInfo.userAction))
                     .value("request", errorInfo.request)
-                    .value("content_language", getContentLangString())
+                    .value("content_language", getContentLanguageString())
+                    .value("content_country", getContentLanguageString())
```
Comment (B0pol) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:389]: I think it's a mistake
```suggestion
                    .value("content_country", getContentCountryString())
```

**Comment by wb9688** [comment]
@B0pol: That's what it's supposed to be, yeah

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 2]
```diff
@@ -377,9 +461,18 @@ private String getUserActionString(final UserAction userAction) {
...
-    private String getContentLangString() {
-        String contentLanguage = PreferenceManager.getDefaultSharedPreferences(this)
+    private String getContentCountryString() {
+        String contentCountry = PreferenceManager.getDefaultSharedPreferences(this)
                 .getString(this.getString(R.string.content_country_key), "none");
+        if (contentCountry.equals(getString(R.string.default_localization_key))) {
+            contentCountry = Locale.getDefault().toString();
+        }
+        return contentCountry;
+    }
```
Comment (B0pol) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:471]: Do not duplicate

```suggestion
    private String getContentCountryString() {
        return Localization.getPreferredContentCountry(this).getCountryCode();
    }
```

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 3]
```diff
@@ -377,9 +461,18 @@ private String getUserActionString(final UserAction userAction) {
...
+
+    private String getContentLanguageString() {
```
Comment (B0pol) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:473]: Same, don't duplicate, just use `Localization.getPreferredLocalization(this).getLocalizationCode();`

But remember that it's useless now. See https://github.com/TeamNewPipe/NewPipe/issues/3089#issuecomment-585832060. So maybe we should gather app language, with `Localization.getAppLocale(this).getLanguage();`

Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:473]: Ah yes. I wrote that code back in October along with some other changes for more debug info regarding the parsed YouTube pages and didn't think of your recent changes when rebasing.

Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:481]: I think `contentCountry` might be useful in the future as soon as we enabled support for it again. I corrected the code but didn't include the content country in the report and also added a field for the app language.

Comment (B0pol) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:481]: Content country != Localization

**Comment by opusforlife2** [comment]
Awesome! I was considering opening an issue regarding the clunky method of going to the crash report to markdown converter every time. This is much better!

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java (hunk 3 above)
Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:474]: Done

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 4]
```diff
@@ -347,7 +380,10 @@ private String buildJson() {
                     .object()
                     .value("user_action", getUserActionString(errorInfo.userAction))
                     .value("request", errorInfo.request)
-                    .value("content_language", getContentLangString())
+                    .value("content_language", getContentLanguageString())
+                    .value("content_country", getContentCountryString())
+                    .value("app_langauge", getAppLanguage())
```
Comment (B0pol) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:385]: remove this line. Typo + duplicate from the one below

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 5]
```diff
@@ -281,6 +279,39 @@ public boolean onOptionsItemSelected(final MenuItem item) {
...
     }
 
+    private void openPrivacyPolicyDialog(final Context context, final String action) {
+        new AlertDialog.Builder(context)
+                .setIcon(android.R.drawable.ic_dialog_alert)
+                .setTitle(R.string.privacy_policy_title)
+                .setMessage(R.string.start_accept_privacy_policy)
+                .setCancelable(false)
+                .setNeutralButton(R.string.read_privacy_policy, (dialog, which) -> {
+                    Intent webIntent = new Intent(Intent.ACTION_VIEW,
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:289]: `final`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 6]
```diff
@@ -365,6 +400,63 @@ private String buildJson() {
         return "";
     }
 
+    private String buildMarkdown() {
+        try {
+            StringBuilder htmlErrorReport = new StringBuilder();
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:405]: `final`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 7]
```diff
@@ -281,6 +279,39 @@ public boolean onOptionsItemSelected(final MenuItem item) {
...
+                .setMessage(R.string.start_accept_privacy_policy)
+                .setCancelable(false)
+                .setNeutralButton(R.string.read_privacy_policy, (dialog, which) -> {
+                    Intent webIntent = new Intent(Intent.ACTION_VIEW,
+                            Uri.parse(context.getString(R.string.privacy_policy_url))
+                    );
+                    context.startActivity(webIntent);
+                })
+                .setPositiveButton(R.string.accept, (dialog, which) -> {
+                    if (action.equals("EMAIL")) { // send on email
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:295]: "an", not "on"

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 8]
```diff
@@ -205,40 +215,28 @@ protected void onCreate(final Bundle savedInstanceState) {
...
+            openPrivacyPolicyDialog(this, "EMAIL");
+        });
 
-                    })
-                    .setNegativeButton(R.string.decline, (dialog, which) -> {
-                        // do nothing
-                    })
-                    .show();
+        copyButton.setOnClickListener((View v) -> {
+            ClipboardManager clipboard =
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:226]: `final`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 9]
```diff
@@ -205,40 +215,28 @@ protected void onCreate(final Bundle savedInstanceState) {
...
+                    (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
+            ClipData clip = ClipData.newPlainText(
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:228]: `final`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java [hunk 10]
```diff
@@ -193,7 +200,10 @@ protected void onCreate(final Bundle savedInstanceState) {
             actionBar.setDisplayShowTitleEnabled(true);
         }
 
-        Button reportButton = findViewById(R.id.errorReportButton);
+        Button reportEmailButton = findViewById(R.id.errorReportEmailButton);
+        Button copyButton = findViewById(R.id.errorReportCopyButton);
+        Button reportGithubButton = findViewById(R.id.errorReportGitHubButton);
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:205]: `final`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java (hunk 5 above)
Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java]: replaced with `ShareUtils.openInBrowser`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java (hunk 8 above)
Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java]: replaced with `ShareUtils.copyToClipboard()`

File: app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java (hunk 7 above)
Comment (opusforlife2) [reply app/src/main/java/org/schabi/newpipe/report/ErrorActivity.java:286]: or "via"/"through"

**Pull request merged**